### ISSUE TITLE
Error 100: Cannot load chart library on iOS

### ISSUE THREAD

**Issue by Synchro** [firehol/netdata#561]
I have latest git master set up on a server. From a desktop machine it all works perfectly, but when I access it from my iPhone it shows "Error 100: Cannot load chart library ... dygraph-combined.js", and none of the graphs load. Next to each section it says "chart library "dygraph" is not enabled", and no charts are drawn. Any idea why that might be?

**Comment by ktsaou** [comment]
Does it do the same on http://london.netdata.rocks ?
Which iOS version?

**Comment by Synchro** [comment]
iOS 9.3.2. It works on that URL. I can request the URL of the script directly and it works. I can also load the site in other browsers on the same phone (e.g. Brave). It's only loading it from within the netdata app in Safari that seems to be a problem. The only other thing I'm doing that's different to stock config is that I have an nginx reverse proxy in front of netdata to add a TLS wrapper, but as I said that's working fine in all other browsers.

I've enabled remote debug on my phone and I can see in the web inspector that it is loading the `dygraph-combined.js` file without any errors, so there must be an issue with using it after it is loaded. I did manage to provoke this error, not sure if it's related:

```
[Error] TypeError: null is not an object (evaluating 'data.person_guid')
	(anonymous function) (dashboard.js:5652)
	i (jquery-1.12.0.min.js:2:27457)
	fireWith (jquery-1.12.0.min.js:2:28217)
	y (jquery-1.12.0.min.js:4:22604)
	c (jquery-1.12.0.min.js:4:26752)
```

**Comment by ktsaou** [comment]
I see this error is triggered if the dashboard cannot talk to the registry. It should be related, I'll fix it though.

Have you checked your nginx error.log?

All requests made should be logged by nginx and netdata access.log. Have you checked that the requests are reaching them?

**Comment by Synchro** [comment]
No errors are logged by nginx. Here's an example load from Safari (it's currently public so you can look at it too):
```
2a02:8434:2a39:2900:9d8c:4c78:fe81:672c - - [17/Jun/2016:13:23:51 +0000] "GET /api/v1/data?chart=system.ram&format=array&points=60&group=average&options=absolute|jsonwrap|nonzero|percentage&after=-60&dimensions=cached|free&_=1466169801169 HTTP/2.0" 200 715 "https://netdata.chamsocial.com/" "Mozilla/5.0 (iPhone; CPU iPhone OS 9_3_2 like Mac OS X) AppleWebKit/601.1.46 (KHTML, like Gecko) Version/9.0 Mobile/13F69 Safari/601.1"
2a02:8434:2a39:2900:9d8c:4c78:fe81:672c - - [17/Jun/2016:13:23:51 +0000] "GET /api/v1/data?chart=system.ipv4&format=array&points=60&group=average&options=absolute|jsonwrap|nonzero&after=-60&dimensions=sent&_=1466169801170 HTTP/2.0" 200 785 "https://netdata.chamsocial.com/" "Mozilla/5.0 (iPhone; CPU iPhone OS 9_3_2 like Mac OS X) AppleWebKit/601.1.46 (KHTML, like Gecko) Version/9.0 Mobile/13F69 Safari/601.1"
2a02:8434:2a39:2900:9d8c:4c78:fe81:672c - - [17/Jun/2016:13:23:51 +0000] "GET /api/v1/data?chart=system.ipv4&format=array&points=60&group=average&options=absolute|jsonwrap|nonzero&after=-60&dimensions=received&_=1466169801171 HTTP/2.0" 200 764 "https://netdata.chamsocial.com/" "Mozilla/5.0 (iPhone; CPU iPhone OS 9_3_2 like Mac OS X) AppleWebKit/601.1.46 (KHTML, like Gecko) Version/9.0 Mobile/13F69 Safari/601.1"
2a02:8434:2a39:2900:9d8c:4c78:fe81:672c - - [17/Jun/2016:13:23:51 +0000] "GET /api/v1/data?chart=system.cpu&format=array&points=60&group=average&options=absolute|jsonwrap|nonzero&after=-60&_=1466169801172 HTTP/2.0" 200 760 "https://netdata.chamsocial.com/" "Mozilla/5.0 (iPhone; CPU iPhone OS 9_3_2 like Mac OS X) AppleWebKit/601.1.46 (KHTML, like Gecko) Version/9.0 Mobile/13F69 Safari/601.1"
2a02:8434:2a39:2900:9d8c:4c78:fe81:672c - - [17/Jun/2016:13:23:51 +0000] "GET /api/v1/data?chart=system.io&format=array&points=60&group=average&options=absolute|jsonwrap|nonzero&after=-60&dimensions=out&_=1466169801173 HTTP/2.0" 200 658 "https://netdata.chamsocial.com/" "Mozilla/5.0 (iPhone; CPU iPhone OS 9_3_2 like Mac OS X) AppleWebKit/601.1.46 (KHTML, like Gecko) Version/9.0 Mobile/13F69 Safari/601.1"
2a02:8434:2a39:2900:9d8c:4c78:fe81:672c - - [17/Jun/2016:13:23:51 +0000] "GET /api/v1/data?chart=system.io&format=array&points=60&group=average&options=absolute|jsonwrap|nonzero&after=-60&dimensions=in&_=1466169801174 HTTP/2.0" 200 527 "https://netdata.chamsocial.com/" "Mozilla/5.0 (iPhone; CPU iPhone OS 9_3_2 like Mac OS X) AppleWebKit/601.1.46 (KHTML, like Gecko) Version/9.0 Mobile/13F69 Safari/601.1"
2a02:8434:2a39:2900:9d8c:4c78:fe81:672c - - [17/Jun/2016:13:23:51 +0000] "GET /api/v1/data?chart=system.swap&format=array&points=60&group=average&options=absolute|jsonwrap|nonzero|percentage&after=-60&dimensions=free&_=1466169801175 HTTP/2.0" 200 525 "https://netdata.chamsocial.com/" "Mozilla/5.0 (iPhone; CPU iPhone OS 9_3_2 like Mac OS X) AppleWebKit/601.1.46 (KHTML, like Gecko) Version/9.0 Mobile/13F69 Safari/601.1"
2a02:8434:2a39:2900:9d8c:4c78:fe81:672c - - [17/Jun/2016:13:23:51 +0000] "GET / HTTP/2.0" 200 29884 "-" "Mozilla/5.0 (iPhone; CPU iPhone OS 9_3_2 like Mac OS X) AppleWebKit/601.1.46 (KHTML, like Gecko) Version/9.0 Mobile/13F69 Safari/601.1"
2a02:8434:2a39:2900:9d8c:4c78:fe81:672c - - [17/Jun/2016:13:23:51 +0000] "GET /dashboard.js?v40 HTTP/2.0" 200 45929 "https://netdata.chamsocial.com/" "Mozilla/5.0 (iPhone; CPU iPhone OS 9_3_2 like Mac OS X) AppleWebKit/601.1.46 (KHTML, like Gecko) Version/9.0 Mobile/13F69 Safari/601.1"
2a02:8434:2a39:2900:9d8c:4c78:fe81:672c - - [17/Jun/2016:13:23:51 +0000] "GET /css/font-awesome.min.css HTTP/2.0" 200 7100 "https://netdata.chamsocial.com/" "Mozilla/5.0 (iPhone; CPU iPhone OS 9_3_2 like Mac OS X) AppleWebKit/601.1.46 (KHTML, like Gecko) Version/9.0 Mobile/13F69 Safari/601.1"
2a02:8434:2a39:2900:9d8c:4c78:fe81:672c - - [17/Jun/2016:13:23:51 +0000] "GET /css/bootstrap.slate.min.css HTTP/2.0" 200 24746 "https://netdata.chamsocial.com/" "Mozilla/5.0 (iPhone; CPU iPhone OS 9_3_2 like Mac OS X) AppleWebKit/601.1.46 (KHTML, like Gecko) Version/9.0 Mobile/13F69 Safari/601.1"
2a02:8434:2a39:2900:9d8c:4c78:fe81:672c - - [17/Jun/2016:13:23:51 +0000] "GET /css/bootstrap-toggle.min.css HTTP/2.0" 200 833 "https://netdata.chamsocial.com/" "Mozilla/5.0 (iPhone; CPU iPhone OS 9_3_2 like Mac OS X) AppleWebKit/601.1.46 (KHTML, like Gecko) Version/9.0 Mobile/13F69 Safari/601.1"
2a02:8434:2a39:2900:9d8c:4c78:fe81:672c - - [17/Jun/2016:13:23:51 +0000] "GET /dashboard.slate.css HTTP/2.0" 200 2761 "https://netdata.chamsocial.com/" "Mozilla/5.0 (iPhone; CPU iPhone OS 9_3_2 like Mac OS X) AppleWebKit/601.1.46 (KHTML, like Gecko) Version/9.0 Mobile/13F69 Safari/601.1"
2a02:8434:2a39:2900:9d8c:4c78:fe81:672c - - [17/Jun/2016:13:23:51 +0000] "GET /lib/jquery-1.12.0.min.js HTTP/2.0" 200 37567 "https://netdata.chamsocial.com/" "Mozilla/5.0 (iPhone; CPU iPhone OS 9_3_2 like Mac OS X) AppleWebKit/601.1.46 (KHTML, like Gecko) Version/9.0 Mobile/13F69 Safari/601.1"
2a02:8434:2a39:2900:9d8c:4c78:fe81:672c - - [17/Jun/2016:13:23:52 +0000] "GET /api/v1/charts?_=1466169831951 HTTP/2.0" 200 13782 "https://netdata.chamsocial.com/" "Mozilla/5.0 (iPhone; CPU iPhone OS 9_3_2 like Mac OS X) AppleWebKit/601.1.46 (KHTML, like Gecko) Version/9.0 Mobile/13F69 Safari/601.1"
```

Notice that it doesn't appear to request `dygraph-combined.js` at all, and it doesn't continue polling for data. I managed to provoke Brave into producing an error 101, failing to load jQuery, but again no error is reported on the server side.

**Comment by ktsaou** [comment]
I fixed the javascript error we were talking before.
Let's see if now it works. Could you please check?

**Comment by Synchro** [comment]
I pulled and recompiled, still getting the same problem. Though that JS error has not reappeared, I did get a different one if I scrolled around a bit:

```
[Error] TypeError: null is not an object (evaluating 'data.urls')
	(anonymous function) (dashboard.js:5656)
	i (jquery-1.12.0.min.js:2:27457)
	fireWith (jquery-1.12.0.min.js:2:28217)
	y (jquery-1.12.0.min.js:4:22604)
	c (jquery-1.12.0.min.js:4:26752)
```

Here's a screen shot from web inspector connected to my phone - you can see it's fetching the `dygraph-combined.js` file (or at least a cached copy), so I don't know why it's complaining - I don't know what the error 100 means.

![screen shot 2016-06-17 at 22 45 35](https://cloud.githubusercontent.com/assets/81561/16164287/8fa5a508-34dd-11e6-8b61-ef0403bbd441.png)

**Comment by ktsaou** [comment]
ok. I fixed that error too.

Error 100 is my code for "cannot load chart library". I raise this error when the jquery ajax call to load a javascript library returns error. I don't know why this happens in your specific case. So, let's fix everything obvious to see what will remain.

Please try again...

**Comment by ktsaou** [comment]
Is it still not working for you?

**Comment by Synchro** [comment]
Sorry for not getting back to you. I've just updated again, and I'm still having the same issue. What's weird is that in the inspector I can see that it's loading the dygraph-combined.js file successfully, and it's pulling it from cache afterwards, but when I compared loading the same file from desktop and phone, I noticed the files are different sizes! On the phone it's truncated at 105k, the desktop version gets 120k. So, it seems the problem is that my phone has a bad truncated copy cached. iOS doesn't have a cache clearing option (well, it has a *nuke everything for all sites including all other browsers on all devices using the same iCloud account* option, but I don't really want to do that!), but by repeatedly hitting the url for the library file I managed to make it load a fresh copy, and then *it worked*.
